src/router.rs

```diff
@@ -108,6 +109,35 @@ impl RouteInfo {
         }
         false
     }
+
+    /// Check if application contains matching route.
+    ///
+    /// This method does take `prefix` into account.
+    /// For example if prefix is `/test` and router contains route `/name`,
+    /// following path would be recognizable `/test/name` and `has_route_prefixed()` call
+    /// would return `true`.
+    pub fn has_route_prefixed(&self, path: &str) -> bool {
+        let path = if path.is_empty() {
+            "/".into()
+        } else if !self.router.prefix.is_empty() {
+            let pb = PathBuf::from(path);
+            if let Ok(p) = pb.strip_prefix(Path::new(&self.router.prefix)) {
+                format!("/{}", p.to_str().unwrap_or(path))
+            } else {
+                path.into()
```

Comment:
Shouldn't we return here false as prefix is missing?

Reply:
But how route can match, if prefix is missing?
Isn't the whole point of this function is to match against prefix and path together? 🤔

Reply:
Check out my updated code, the initial issue asked to be able to match both in a streamlined fashion, it should be clearer to reason about